apps/teams-test-app/src/components/PagesAPIs.tsx

```diff
@@ -126,12 +127,21 @@ const RegisterFocusEnterHandler = (): React.ReactElement =>
   ApiWithoutInput({
     name: 'registerFocusEnterHandler',
     title: 'Register On Focus Enter Handler',
-    onClick: async setResult => {
-      teamsCore.registerFocusEnterHandler(navigateForward => {
-        setResult('successfully called with navigateForward:' + navigateForward);
-        return true;
-      });
-      return 'registered';
+    onClick: {
+      withPromise: async setResult => {
+        teamsCore.registerFocusEnterHandler(navigateForward => {
+          setResult('successfully called with navigateForward:' + navigateForward);
+          return true;
+        });
+        return 'registered';
+      },
+      withCallback: setResult => {
+        registerFocusEnterHandler(navigateForward => {
+          setResult('successfully called with navigateForward:' + navigateForward);
+          return true;
+        });
+        return 'registered';
```

Comment:
Will this output correctly or does this need to be setResult('registered')?

Reply:
I guess I need to clean up my mind, thanks!